### PR TITLE
move sign in labels to be above inputs

### LINKED ISSUE #28696: Update the UI for the sign in screen
### Feature Description

The sign-in page's form layout is currently different from the other forms and inputs in Gitea's i.e. `/user/settings`. To maintain consistency, we should update the sign-in form to have the labels above the input. Additionally, many users find it easier to log in using OAuth. As a result, I suggest that we move these options to be above the sign-in form.

We should also update the following forms in separate PRs

- /user/sign_up
- /user/forgot_password
- /user/login/openid

### Screenshots

A mock of the direction I'm proposing.

stacked

### PR DESCRIPTION
**Pull request by rafh**
There are a few inconsistencies within Gitea and this PR addresses one of them. This PR updates the sign-in page layout, including the register and openID tabs, to match the layout of the settings pages (`/user/settings`) for more consistency. 

**Before**
Screenshot 2024-02-05 at 8 27 24 AM

**After**
Screenshot 2024-02-05 at 8 26 39 AM

### CODE REVIEW

**Comment by KN4CK3R** [comment]
> OAuth buttons have been moved above the sign-in form as OAuth is typically used more than traditional sign-in forms

That's debatable 😏 I like the other order more. A non-representative survey between me and some online services results in first login form then other. (GitLab, Instagram, Firefox, Cloudflare = first login form, Twitter/X = first oauth)

**Comment by denyskon** [comment]
@KN4CK3R I agree, OAuth as second is the more common I think.

**Comment by rafh** [comment]
> > OAuth buttons have been moved above the sign-in form as OAuth is typically used more than traditional sign-in forms
> 
> That's debatable 😏 I like the other order more. A non-representative survey between me and some online services results in first login form then other. (GitLab, Instagram, Firefox, Cloudflare = first login form, Twitter/X = first oauth)

I can see how the order might feel more familiar with the login form first. However, research actually shows a strong shift towards passwordless options like social logins and SSO. For example, this [report](https://www.loginradius.com/blog/identity/social-login-infographic/ ) shows many consumers prefer social logins (LoginRadius Consumer Digital Identity Trend Report 2022). The 2023 report isn't published yet but you can download it [here](https://www.loginradius.com/resource/consumer-digital-identity-trend-report-2023). This inforgraphic from this report shows the trends towards passwordless.

Screenshot 2024-01-11 at 9 51 27 AM

I understand personal preference is subjective, but the data seems to favor the convenience and security of passwordless sign-in. Finally, many sites are using the proposed layout first, perhaps due to its increasing popularity. Here are a few examples:

[logrocket](https://app.logrocket.com/)
[medium](https://medium.com/)
[dropbox](https://www.dropbox.com/login)
[fitbit](https://accounts.fitbit.com/login)
[strava](https://www.strava.com/login)
[spotify](https://accounts.spotify.com/en/login)

**Comment by denyskon** [comment]
I think there's a big difference between developers or at least technical users who will use Gitea, and a 'normal consumer' on a site like Spotify, however I won't block this change....

**Comment by rafh** [comment]
> I think there's a big difference between developers or at least technical users who will use Gitea, and a 'normal consumer' on a site like Spotify, however I won't block this change....

I understand your point and I agree there's a difference. However, we have to keep in mind that not all users are developers. Gitea is an open-source project that caters to a variety of end users.

**Comment by lunny** [comment]
If no maintainers support this PR, I think maybe we can close this one. Let's wait for more reviews.

**Comment by KN4CK3R** [comment]
> If no maintainers support this PR, I think maybe we can close this one. Let's wait for more reviews.

The moved input field labels are good.

> However, research actually shows a strong shift towards passwordless options like social logins and SSO. For example, this [report](https://www.loginradius.com/blog/identity/social-login-infographic/) shows many consumers prefer social logins (LoginRadius Consumer Digital Identity Trend Report 2022).

LoginRadius offers services related to SSO. Not really surprising they have a report that shows what they want to sell.

**Comment by rafh** [comment]
@KN4CK3R I've updated this PR to only move the labels above the inputs. What are your thoughts?

**Review by denyskon — APPROVED** [review]

**Review by KN4CK3R — APPROVED** [review]

**Comment by lafriks** [comment]
Maybe remove tabs at all (there is already link to register) and openid can be moved as other external buttons

**Comment by kdumontnu** [comment]
> Maybe remove tabs at all (there is already link to register) and openid can be moved as other external buttons

I totally agree with this, but it should be a separate PR I think (no reason to increase scope of this PR and miss 1.22 feature freeze for example)

**Comment by denyskon** [comment]
It isn't a "feature" imo, so it can be merged after the freeze

**Comment by wxiaoguang** [comment]
I guess it breaks some other pages.

image


image

**Comment by rafh** [comment]
I'll open a PR to fix these issues.

**Comment by lunny** [comment]
This change also breaks captcha, since it broke two places. I will revert this one and please send a new PR.

**Comment by rafh** [comment]
> This change also breaks captcha, since it broke two places. I will revert this one and please send a new PR.

I'll open a new PR to handle the captcha. The Gitea frontend has many different scenarios and could benefit from snapshot/visual testing for various scenarios. It won't be part of my next PR but I think it's worth considering.

**Pull request merged**
